libs/beacon/src/beacon_service.cpp

```diff
@@ -95,14 +99,56 @@ BeaconService::BeaconService(MuddleInterface &               muddle,
 
 BeaconService::Status BeaconService::GenerateEntropy(uint64_t block_number, BlockEntropy &entropy)
 {
-  FETCH_LOG_DEBUG(LOGGING_NAME, "Requesting entropy for block number: ", block_number);
+  FETCH_LOG_TRACE(LOGGING_NAME, "Requesting entropy for block number: ", block_number);
   beacon_entropy_last_requested_->set(block_number);
 
   FETCH_LOCK(mutex_);
 
-  if (completed_block_entropy_.find(block_number) != completed_block_entropy_.end())
+#if 1
+  {
+    static std::string last_have{};
```

Comment:
Suggest thread_local or class variable

Reply:
Although this looks like debug - could be replaced by a telemetry gague 'beacon_entropy_last_supplied'